modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -3190,14 +3190,7 @@ protected static String getEnumValueForProperty(
     }
 
     protected void SortModelPropertiesByRequiredFlag(CodegenModel model) {
-        Comparator<CodegenProperty> comparator = new Comparator<CodegenProperty>() {
-            @Override
-            public int compare(CodegenProperty one, CodegenProperty another) {
-                if (one.required == another.required) return 0;
-                else if (one.required) return -1;
-                else return 1;
-            }
-        };
+        Comparator<CodegenProperty> comparator = Comparator.comparing(prop -> !prop.required);
```

Comment:
since this touches default codegen (which requires a bit more time to review from the core team), can you please file this change separately in another PR?